lib/reek/source/source_locator.rb

```diff
@@ -11,60 +12,56 @@ class SourceLocator
       #
       # paths - a list of paths as Strings
       def initialize(paths)
-        @pathnames = paths.
-          map { |path| Pathname.new(path.chomp('/')) }.
-          flat_map { |pathname| current_directory?(pathname) ? pathname.entries : pathname }
+        @paths = paths.map(&method(:Pathname)).flat_map do |path|
+          current_directory?(path) ? path.entries : path
+        end
       end
 
       # Traverses all paths we initialized the SourceLocator with, finds
       # all relevant ruby files and returns them as a list.
       #
-      # @return [Array<File>] - Ruby files found
+      # @return [Array<Pathname>] - Ruby paths found
```

Comment:
Yes, this is a good application of the SRP. See my other comment on letting `CLI::Input` convert the result into `File` objects.